AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3657,6 +3658,37 @@ function Debug-AzStorageAccountAuth {
             }
         }
 
+        if (!$filterIsPresent -or $Filter -match "CheckDefaultSharePermission")
+        {
+            try {
+                $checksExecuted += 1
+                Write-Verbose "CheckDefaultSharePermission - START"
+
+                $StorageAccountObject = Validate-StorageAccount `
+                    -ResourceGroupName $ResourceGroupName `
+                    -StorageAccountName $StorageAccountName `
+                    -ErrorAction Stop
+ 
+                Write-Verbose -Message "Storage account $StorageAccountName is already joined in domain $($activeDirectoryProperties.DomainName)."
+                
+                $DefaultSharePermission = $StorageAccountObject.AzureFilesIdentityBasedAuth.DefaultSharePermission
+                
+                # If DefaultSharePermission is null or 'None'
+                if((!$DefaultSharePermission) -or ($DefaultSharePermission -eq 'None')){
+                    Write-Error "CheckDefaultSharePermission - FAILED"
+                    $DefaultSharePermission = "Not Configured"
+                }else{
+                    Write-Verbose "CheckDefaultSharePermission - SUCCESS"

```

Comment:
The check should be SUCESS whether DefaultSharePermission is null or not since either value is valid.